src/components/SideBar/SideBar.jsx

```diff
@@ -0,0 +1,53 @@
+import { useState, useEffect, useContext } from "react";
+import { useNavigate } from "react-router-dom";
+import "./SideBar.css";
+import { Link } from "react-router-dom";
+import { AuthContext } from "../../contexts/spotifyContext";
+
+function SideBar() {
+  //const [accessToken, setAccessToken] = useState(null);
+  const [userData, setUserData] = useState({
+    display_name: "",
+    images: [],
+  });
+  const accessToken = localStorage.getItem("spotify_access_token");
+  useEffect(() => {
+    // const accessToken = localStorage.getItem("spotify_access_token");
+    console.log(accessToken);
+    if (accessToken) {
+      fetch("https://api.spotify.com/v1/me", {
+        headers: {
+          Authorization: `Bearer ${accessToken}`,
+        },
+      })
+        .then((res) => res.json())
+        .then((data) => {
+          setUserData({
+            display_name: data.display_name,
+            images: data.images,
+          });
+        })
+        .catch((err) => console.error("Error fetching user data:", err));
```

Comment:
You should **not** use `catch` in `Api` because you need to `catch` possible errors at the very **end** of any request, and this end is in (`App.js`) after all `then` blocks. And in real projects the `catch` blocks usually open error popups and so on.

Please, remove it from `Api` and add it in `index.js`.